### PR TITLE
Left panel lifting.

### PR DESCRIPTION
**Pull request by pieer**
Update the left menu

### CODE REVIEW

**Comment by taoeffect** [comment]
There's something weird going on when I tried to test this PR. I don't know if it has to do with this PR or a previous PR that I merged, but testing inviting users to different groups between Safari and Firefox seems broken. This is mostly a note to myself to remember to re-test this and figure out what's going on later today.

File: frontend/views/components/ListItem.vue [hunk 1]
```diff
@@ -76,31 +67,39 @@ export default {
 .c-item-link {
   display: flex;
   align-items: center;
-  padding: 0 $spacer;
-  height: $spacer-lg;
+  padding: 0 1.25rem;
```
Comment (sandrina-p) [inline frontend/views/components/ListItem.vue]: shouldn't it better to use `$spacer*1.25` so we avoid raw values?

Comment (pieer) [reply frontend/views/components/ListItem.vue]: Please no

File: frontend/views/components/ListItem.vue [hunk 2]
```diff
@@ -76,31 +67,39 @@ export default {
...
   color: $text_0;
+  font-weight: 100;
+  position: relative;
 
   i {
     width: 1rem;
-    margin-right: $spacer-md;
-    font-size: 0.77rem;
-    color: $general_0;
+    margin-right: 1.1rem;
```
Comment (sandrina-p) [inline frontend/views/components/ListItem.vue]: why not keeping `$spacer`?

File: frontend/views/components/ListItem.vue (hunk 1 above)
Comment (taoeffect) [reply frontend/views/components/ListItem.vue]: Ah come on @pieer, you can't just resolve feedback by saying "no", there needs to be some sort of reasoning behind that "no" spelled out. I don't understand the intricacies of CSS here but could it be that you're saying "no" because it's already a relative unit (rem)?

Comment (pieer) [reply frontend/views/components/ListItem.vue]: ok.

Comment (sandrina-p) [reply frontend/views/components/ListItem.vue]: I'm still wondering why no, really. $spacer is 1rem, so this a relative size to it (*1.25)... it's the same as not using $spacer and replace it by 1rem.

-- edit -- sorry caps, wasn't on purpose.

Comment (pieer) [reply frontend/views/components/ListItem.vue]: If you want..

**Comment by sandrina-p** [comment]
I believe the development isn't finished yet, right @pieer ? Could you explicit say it on the description by mention what's missing and adding a WIP to the title.

**Review by mmbotelho — CHANGES_REQUESTED** [review]
Great work on this one! This is looking great, and I love all the animations you added. However, there are some small details that are not according to design. Here's a list of what I found:

- Hover, active and focused state are not as specified in the design system. They should look like this:
Captura de ecrã 2019-08-28, às 15 47 59

- The **notifications** and **settings** buttons should be `button.is-icon-small`;

- The "add group" button (icon) does not have the correct CSS:
Captura de ecrã 2019-08-28, às 16 04 44

- I noticed that the `button.is-icon.has-background` does not have the correct hover styles in the design-system page. Even though this is not part of the PR, the correct styles can be found [here (Figma)](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=48%3A26)
Captura de ecrã 2019-08-28, às 16 16 13

- I did not include comments about the tablet and mobile versions of the menu - These can be addressed in a new issue/pull request, since I still need to design them.

- The **Add a group** flow is not in a full page modal [as designd (Figma)](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=861%3A36) but I'm assuming that this will be done in a different PR as well.

Thank you!

**Comment by mmbotelho** [comment]
Another thing I noticed is that the sidebar does not show the group avatar I uploaded. Is this also supposed to be done in another PR?

File: frontend/assets/style/components/_forms.scss [hunk 3]
```diff
@@ -129,10 +129,6 @@ textarea,
     border-right: 1px solid $general_0;
   }
 
-  * + .input {
-    border: 0;
```
Comment (sandrina-p) [inline frontend/assets/style/components/_forms.scss:133]: In which scenarios this can happen/be helpful?
It would be nice to leave a comment in the code explaining that since this selector is global...

File: frontend/views/components/ListItem.vue [hunk 4]
```diff
@@ -76,31 +67,39 @@ export default {
...
+  font-weight: 100;
+  position: relative;
 
   i {
-    width: 1rem;
-    margin-right: $spacer-md;
-    font-size: 0.77rem;
-    color: $general_0;
+    width: $spacer;
+    margin-right: 1.2rem;
```
Comment (sandrina-p) [inline frontend/views/components/ListItem.vue]: Based on the last Monday call, should this value be 1rem to stick with the current spacing values? (I checked Figma and it's 16px indeed, not 19px)

File: frontend/views/containers/sidebar/GroupsList.vue [hunk 5]
```diff
@@ -1,60 +1,33 @@
...
+    v-for='(group, index) in groupsByName'
+    :key='`group-${index}`'
+    tag='button'
+    :class="{ 'is-active': currentGroupId === group.contractID}"
+  )
+    avatar.c-avatar(
+      src='/assets/images/default-avatar.png'
+      :alt='group.groupName'
+      :blobURL='group.groupPicture'
+      @click="handleMenuSelect(group.contractID)"
```
Comment (sandrina-p) [inline frontend/views/containers/sidebar/GroupsList.vue]: Each group is a clickable element, so it should be a `button` or an anchor if we redirect to another page. Since at the moment we keep the same URL, I believe a `button` is the best.

File: frontend/views/containers/sidebar/GroupsList.vue [hunk 6]
```diff
@@ -81,42 +53,82 @@ export default {
...
+      animation: spin 0.5s ease-out;
+    }
   }
 
-  &:hover,
-  &:focus {
-    box-shadow: none;
+  &:not(.is-active) {
+    cursor: pointer;
+    &:hover::before {
```
Comment (sandrina-p) [inline frontend/views/containers/sidebar/GroupsList.vue:101]: Don't forget `:focus` state as part of possible user interactions. (Please review the other places too where there's only `:hover`)

**Review by sandrina-p — CHANGES_REQUESTED** [review]

File: frontend/views/containers/sidebar/GroupsList.vue (hunk 5 above)
Comment (sandrina-p) [reply frontend/views/containers/sidebar/GroupsList.vue]: Also, this doesn't seem to work properly. When I click, I need to refresh the page so it really changes the group dashboard, otherwise, the group scope is still the previous.

File: frontend/views/containers/sidebar/GroupsList.vue [hunk 7]
```diff
@@ -1,60 +1,33 @@
...
-      i.icon-angle-down
+ul.c-group-list(v-if='groupsByName.length')
+  li.c-group-list-item.group-badge(
+    v-for='(group, index) in groupsByName'
+    :key='`group-${index}`'
+    tag='button'
+    :class="{ 'is-active': currentGroupId === group.contractID}"
+  )
+    avatar.c-avatar(
+      src='/assets/images/default-avatar.png'
```
Comment (sandrina-p) [inline frontend/views/containers/sidebar/GroupsList.vue]: > Another thing I noticed is that the sidebar does not show the group avatar I uploaded. Is this also supposed to be done in another PR?

It must be a bug @mmbotelho yes, here's the cause of the problem: The source image is a static one at the moment. It's a easy fix for @pieer ;)

**Comment by taoeffect** [comment]
@pieer - heads up, a recent merge means there's now a conflict

File: frontend/assets/style/components/_forms.scss (hunk 3 above)
Comment (pieer) [reply frontend/assets/style/components/_forms.scss:133]: Sorry, I'm not sure to follow..

File: frontend/views/containers/sidebar/GroupsList.vue (hunk 7 above)
Comment (pieer) [reply frontend/views/containers/sidebar/GroupsList.vue]: Ths static file is for the default in case there is no image. I believe the avatar is another issue that you're working on?

File: frontend/views/containers/sidebar/GroupsList.vue (hunk 5 above)
Comment (pieer) [reply frontend/views/containers/sidebar/GroupsList.vue]: The group is changing correctly in the vuex store. It also update the group in the selector component. There must be another problem coming from other components listeners.

**Comment by mmbotelho** [comment]
Missing tooltips with group names and button label:
Captura de ecrã 2019-08-29, às 22 36 25
Captura de ecrã 2019-08-29, às 22 35 45
Captura de ecrã 2019-08-29, às 22 35 58

**Review by mmbotelho — CHANGES_REQUESTED** [review]
Missing tooltips with group names and button label:

Captura de ecrã 2019-08-29, às 22 36 25
Captura de ecrã 2019-08-29, às 22 35 45
Captura de ecrã 2019-08-29, às 22 35 58

**Review by mmbotelho — DISMISSED** [review]
Awesome! Great job! 😄

File: frontend/assets/style/components/_forms.scss (hunk 3 above)
Comment (sandrina-p) [reply frontend/assets/style/components/_forms.scss:133]: Nevermind, when I saw this I thought you added it, not removed it. Pardon the confusion

File: frontend/views/containers/sidebar/GroupsList.vue (hunk 7 above)
Comment (sandrina-p) [reply frontend/views/containers/sidebar/GroupsList.vue]: I'm working on the avatar issue but it's on profile avatar, not group avatar. It's different things :)

File: frontend/views/containers/sidebar/GroupsList.vue [hunk 8]
```diff
@@ -1,70 +1,50 @@
...
-    Avatar
+    Avatar,
+    Tooltip
   },
   computed: {
     ...mapState([
       'currentGroupId'
     ]),
     ...mapGetters([
       'groupSettings',
```
Comment (sandrina-p) [inline frontend/views/containers/sidebar/GroupsList.vue]: I believe groupSettings isn't needed now.

File: frontend/views/containers/sidebar/GroupsList.vue (hunk 7 above)
Comment (pieer) [reply frontend/views/containers/sidebar/GroupsList.vue]: Is it?

File: frontend/views/containers/sidebar/GroupsList.vue [hunk 9]
```diff
@@ -1,70 +1,50 @@
...
+    :class="{ 'is-active': currentGroupId === group.contractID}"
+  )
+    tooltip(
+      direction="right"
+      :text='group.groupName'
+    )
+      button.is-unstyled(@click="handleMenuSelect(group.contractID)")
+        avatar.c-avatar(
+          src='/assets/images/default-avatar.png'
+          :blobURL='group.groupPicture'
```
Comment (sandrina-p) [inline frontend/views/containers/sidebar/GroupsList.vue]: @pieer the cause of the missing group picture is caused by how "groupsByName" is constructed.

Before we accessed `groupSettings` that brings all the settings of a group. Now we just consume `groupsByName` that doesn't bring the `groupPicture`. We need to change `state.js`  so it also returns the group settings so we can access it as `group.settings.groupPicture`.

**Review by sandrina-p — DISMISSED** [review]
It's great now @pieer . Just need to fix that groupPicture thing and it's good to go!

File: frontend/views/containers/sidebar/GroupsList.vue (hunk 9 above)
Comment (pieer) [reply frontend/views/containers/sidebar/GroupsList.vue]: I see! ok, it's fix now

File: frontend/model/state.js [hunk 10]
```diff
@@ -169,7 +169,11 @@ const getters = {
     // due to the same flow issue as https://github.com/facebook/flow/issues/5838
     return Object.keys(contracts)
       .filter(contractID => contracts[contractID].type === 'group' && state[contractID].settings)
-      .map(contractID => ({ groupName: state[contractID].settings.groupName, contractID }))
+      .map(contractID => ({
+        groupName: state[contractID].settings.groupName,
+        contractID,
+        groupPicture: state[contractID].settings.groupPicture
```
Comment (taoeffect) [inline frontend/model/state.js]: So the reason why this getter returned just `groupName` and `contractID` is because with that information, you can get everything else you need for each group.

This function is not designed to return everything you might want about all groups. It is designed as a hook into knowledge about all groups, that can be used as such to the degree required.

So there is a danger her of unnecessarily continuing to overload this function with more and more information that is not needed 99% of the time it is used. That is creates a problem because this function is used in many places, and so we would be repeatedly constructing massive objects and passing around that data to places that don't ever make use of that data. In a word, it is wasteful.

So I strongly recommend that instead of taking the approach of adding `groupPicture` to this returned object, that you instead use `group.contractID` (in GroupsList.vue below) to retrieve the data you need.

So this getter would be restored to how it was, and down below it would read something like `$store.state[group.contractID].settings.groupPicture`

**Review by taoeffect — CHANGES_REQUESTED** [review]
On change requested, otherwise looks great!: https://github.com/okTurtles/group-income-simple/pull/636/files#r319631047

**Comment by taoeffect** [comment]
Screen Shot 2019-08-30 at 2 43 06 PM

Scrollbar visible when it shouldn't be

**Comment by taoeffect** [comment]
Screen Shot 2019-08-30 at 2 46 44 PM

```
[Vue warn]: Error in render: "TypeError: member is undefined"

found in

---> <GroupMembers>
       <Page>
         <GroupDashboard>
           <Root> vue.common.dev.js:630:14
TypeError: "member is undefined"
    D main.js:10044
    renderList VueJS
    __vue_render__$D main.js:10037
    VueJS 47
vue.common.dev.js:1893:12
```

## How to reproduce

1. In Firefox Dev Edition, in Container Tab 1, create user `greg` and Group 1, **with** a group picture
2. In Container Tab 2, create user `pierre` and Group 2, **without** a group picture
3. In Container Tab 2, invite `greg` to Group 2
4. In Container Tab 1, accept invite to Group 2 (by clicking the invisible Mailbox link in the Navbar, and copy/pasting the URL in the invite message)

**Review by taoeffect — APPROVED** [review]
Approved, however there's a serious bug in GroupMembers.vue that needs to be fixed: #648
